test/test_json_regex.py

```diff
@@ -0,0 +1,55 @@
+import re
+import unittest
+from json import loads, dumps
+
+json_re = re.compile(r'^(?:["[{]|(?:-?[1-9]\d*(?:\.\d+)?|null|true|false)$)')
+test_case_loadable = (
+    True,
+    False,
+    None,
+    123,
+    -123,
+    3.14,
+    -3.14,
+    {'foo': 'bar'},
+    [1, 2, 3],
+    "foo bar"
+)
+
+test_case_not_loadable = (
+    "/foobar.py",
+    "text",
+    "yes",
+    "0123123",
+)
+
+def try_loads_json(json_str):
+    try:
+        data = loads(json_str)
+    except ValueError:
+        data = None
+    return data
+
+
+class TestJsonRegex(unittest.TestCase):
+
+    def test_loadable_match(self):
+        for i, case in enumerate(test_case_loadable):
+            with self.subTest(i=i):
+                self.assertTrue(json_re.match(dumps(case))is not None, msg="Testing:{}".format(case))
```

Comment:
You need to test `env_value_to_dict` function, not local regexp in this file.

Reply:
It's not possible to test if regex works by only testing `env_value_to_dict` function

Reply:
Nether form the type value of return nor if the return value is same with our test case, if we still using test case like

Reply:
For example, if we using test case ` "foo bar"`(match regex) an testcase `"yes"`(not match with regex), both will return a `string` and both will make the return of `env_value_to_dict` equal with the parameter of this function, which makes us cannot tell if the regex real works or not.

Reply:
Hmm.. you can import the compiled reg exp and test that directly, but also test the env_value_to_dict function.

That said, with `{'TEST1': '"foo bar"', 'TEST2': "yes"}` (first case and the second), you will get back `"foo bar"` and `"yes"`, where first is not the same as the input (one set of quotes is removed).

Reply:
Ok i see the point